### PR TITLE
Feature/consensus protocol version

### PR DESCRIPTION
**Pull request by minsulee2**
Changes:
- Add consensus protocol version
- Check consensus version when messaging
- Encapsulate consensus message

### CODE REVIEW

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

File: consensus/index.js [hunk 1]
```diff
@@ -175,12 +180,40 @@ class Consensus {
     logger.debug(`[${LOG_HEADER}] proposer for epoch ${this.state.epoch}: ${this.state.proposer}`);
   }
 
+  checkConsensusProtocolVersion(msg) {
+    const LOG_HEADER = 'checkConsensusProtocolVersion';
+    const consensusProtoVer = _.get(msg, 'consensusProtoVer');
+    if (!consensusProtoVer || !semver.valid(consensusProtoVer)) {
+      logger.error(`[${LOG_HEADER}] CONSENSUS_PROTOCOL_VERSION cannot be empty.`);
```
Comment (platfowner) [inline consensus/index.js:187]: empty or invalid?

**Review by platfowner — APPROVED** [review]
LGTM with some minor comments.
Thanks!

File: consensus/index.js (hunk 1 above)
Comment (minsulee2) [reply consensus/index.js]: Done!!

**Comment by minsulee2** [comment]
Thank you guys for the review!

**Pull request merged**